test/court/court-settle.js

```diff
@@ -0,0 +1,790 @@
+  ]
+
+  beforeEach('create court', async () => {
+    courtHelper = buildHelper()
+    court = await courtHelper.deploy()
+    voting = courtHelper.voting
+  })
+
+  describe('settle', () => {
+    context('when the given dispute exists', () => {
+      let disputeId, voteId
+      const draftTermId = 4, jurorsNumber = 3
+
+      beforeEach('activate jurors and create dispute', async () => {
+        await courtHelper.activate(jurors)
+
+        await courtHelper.setTerm(1)
+        disputeId = await courtHelper.dispute({ jurorsNumber, draftTermId, disputer })
+        await courtHelper.passTerms(draftTermId - 1) // court is already at term one
+      })
+
+      context('when the given round is valid', () => {
+        const roundId = 0
+        const voters = [
+          { address: juror1000, weight: 1, outcome: OUTCOMES.LOW },
+          { address: juror2000, weight: 1, outcome: OUTCOMES.HIGH },
+          { address: juror4000, weight: 1, outcome: OUTCOMES.LOW },
+        ]
+
+        const itIsAtState = (roundId, state) => {
+          it(`round is at state ${state}`, async () => {
+            const { roundState } = await courtHelper.getRound(disputeId, roundId)
+            assert.equal(roundState.toString(), state, 'round state does not match')
+          })
+        }
+
+        const itFailsToExecuteRuling = () => {
+          it('fails to execute ruling', async () => {
+            await assertRevert(court.executeRuling(disputeId), 'CT_INVALID_ADJUDICATION_STATE')
+          })
+        }
+
+        const itFailsToSettlePenalties = (roundId) => {
+          it('fails to settle penalties', async () => {
+            await assertRevert(court.settlePenalties(disputeId, roundId, jurorsNumber), 'CT_INVALID_ADJUDICATION_STATE')
+          })
+        }
+
+        const itFailsToSettleRewards = (roundId) => {
+          it('fails to settle rewards', async () => {
+            await assertRevert(court.settleReward(disputeId, roundId, juror1000), 'CT_ROUND_PENALTIES_NOT_SETTLED')
+          })
+        }
+
+        const itFailsToSettleAppealDeposits = (roundId) => {
+          it('fails to settle appeal deposits', async () => {
+            await assertRevert(court.settleAppealDeposit(disputeId, roundId), 'CT_ROUND_PENALTIES_NOT_SETTLED')
+          })
+        }
+
+        const itFailsToSettleAll = (roundId) => {
+          itFailsToSettlePenalties(roundId)
+          itFailsToSettleRewards(roundId)
+          itFailsToSettleAppealDeposits(roundId)
+        }
+
+        const itExecutesFinalRulingProperly = expectedFinalRuling => {
+          describe('executeRuling', () => {
+            it('emits an event', async () => {
+              const receipt = await court.executeRuling(disputeId)
+
+              assertAmountOfEvents(receipt, 'RulingExecuted')
+              assertEvent(receipt, 'RulingExecuted', { disputeId, ruling: expectedFinalRuling })
+            })
+
+            it('executes the associated airbitrable', async () => {
```

Comment:
even though 'airbitrable' sounds amazing... 😉

```suggestion
            it('executes the associated arbitrable', async () => {
```